api/openapi.yaml

```diff
@@ -56,6 +56,27 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/ApiItem"
+    post:
+      tags:
+        - "apis"
+      summary: "Deploy new API"
+      description: "Deploys a new API to the cluster"
+      operationId: deployApi
+      requestBody:
+        description: "API content that needs to be deployed"
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                api:
+                  type: object
+              required:
+                - api
+      responses:
```

Comment:
add an error response for if the API is "malformed" - i.e. 
- isn't an OpenAPI 3.x definition
- doesn't contain the required x-kusk extensions for it to be deployable